### PR TITLE
Ensure _acl is updated when _rperm and _wperm updated

### PR DESCRIPTION
**Pull request by steven-supersolid**
This should fix the bug introduced here: https://github.com/ParsePlatform/parse-server/pull/2021/files#diff-82f0219ca721372e8f2bdbc45c83bbdcR303

Updated existing test and added new test.

@drew-gross can you take a look please?

### CODE REVIEW

File: src/Adapters/Storage/Mongo/MongoTransform.js [hunk 1]
```diff
@@ -330,18 +330,11 @@ const parseObjectToMongoObjectForCreate = (className, restCreate, schema) => {
 // Main exposed method to help update old objects.
 const transformUpdate = (className, restUpdate, parseFormatSchema) => {
   let mongoUpdate = {};
-  let acl = addLegacyACL(restUpdate)._acl;
```
Comment (steven-supersolid) [inline src/Adapters/Storage/Mongo/MongoTransform.js]: ._acl is not the full acl but just the legacy part, so none of the properties below exist as we are trying to reference acl._acl._acl for example.
$set._rperm and _wperm are provided via transformKeyValueForUpdate on line 340 so all that needs to be added here is the legacy property _acl

**Comment by steven-supersolid** [comment]
Fixes #2628

File: src/Adapters/Storage/Mongo/MongoTransform.js [hunk 2]
```diff
@@ -330,18 +330,11 @@ const parseObjectToMongoObjectForCreate = (className, restCreate, schema) => {
...
-  if (acl) {
-    mongoUpdate.$set = {};
-    if (acl._rperm) {
-      mongoUpdate.$set._rperm = acl._rperm;
-    }
-    if (acl._wperm) {
-      mongoUpdate.$set._wperm = acl._wperm;
-    }
-    if (acl._acl) {
-      mongoUpdate.$set._acl = acl._acl;
```
Comment (flovilmart) [inline src/Adapters/Storage/Mongo/MongoTransform.js]: Actually, the bug should be fixed by just setting `acl` Instead of `acl._acl`

Comment (steven-supersolid) [inline src/Adapters/Storage/Mongo/MongoTransform.js]: An alternative fix would be to remove the ._acl on line 334 and keep the original code, i.e.
```
  let acl = addLegacyACL(restUpdate);
  if (acl) {
    mongoUpdate.$set = {};
    if (acl._rperm) {
      mongoUpdate.$set._rperm = acl._rperm;
    }
    if (acl._wperm) {
      mongoUpdate.$set._wperm = acl._wperm;
    }
    if (acl._acl) {
      mongoUpdate.$set._acl = acl._acl;
    }
  }
```
I felt that was pointless though as _rperm and _wperm are set elsewhere and thought there must have been a reason line 334 was changed in the first place to reference _acl

Comment (flovilmart) [inline src/Adapters/Storage/Mongo/MongoTransform.js]: If it was written like that, there must be a reason, that is not obvious. The `acl._acl` is clearly an error from the original commiter and not his intention. I'd like to keep the code as it is, as it's working for the other use cases. You don't really know what you maybe breaking by removing it

**Comment by facebook-github-bot-2** [comment]
@steven-supersolid updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2701/files/cde2755b46367fb420b9572104c74ca98b2766fa..cde907f81d8443791a8fd99ced6dfe9826d3dc42)

File: src/Adapters/Storage/Mongo/MongoTransform.js (hunk 1 above)
Comment (flovilmart) [inline src/Adapters/Storage/Mongo/MongoTransform.js]: that will render the following lines incorrect...

Comment (steven-supersolid) [inline src/Adapters/Storage/Mongo/MongoTransform.js]: If you are suggesting not changing line 333 then none of the if clauses will execute on lines 336, 339, 342

The object returned by addLegacyACL has the format:
```
{
 _acl: {
   id: w,
   id: r},
 _rperm: [],
_wperm: []
}
```

Comment (flovilmart) [inline src/Adapters/Storage/Mongo/MongoTransform.js]: So the whole thing as a no-op in the end and now it's breaking all the tests...

**Comment by facebook-github-bot-1** [comment]
@steven-supersolid updated the pull request - [view changes](https://github.com/ParsePlatform/parse-server/pull/2701/files/cde907f81d8443791a8fd99ced6dfe9826d3dc42..b1f9882e358e5777c8aa840fd0bf42a591d60d0b)

File: src/Adapters/Storage/Mongo/MongoTransform.js (hunk 1 above)
Comment (steven-supersolid) [inline src/Adapters/Storage/Mongo/MongoTransform.js]: Sorry I thought it was an intermittent test failure.
The tests were failing when restUpdate contained non-acl updates, e.g. `{ _failed_login_count: { __op: 'Increment', amount: 1 } }`.
I've restored the original code for line 334 which fixes the issue.

Comment (flovilmart) [inline src/Adapters/Storage/Mongo/MongoTransform.js]: Looks good now!

**Pull request merged**